### PR TITLE
[dnm] Menu rework

### PR DESCRIPTION
**Pull request by Nauna**
changes in ApplicationController
-  mementoChange() is what is called on login of backOffice - updated this call to include the redundancy check from function findFirstMenuIHavePermissionFor(dao)
- menuNull property. Need control for times when we want to be in a bad menu state. Sign out clears the menu and memento, therefore need to not try to push anything. Setup is SignOut->menuNull = true, SignIn->menuNull = false.

addresses https://nanopay.atlassian.net/browse/NP-4876

### CODE REVIEW

File: src/menus [hunk 1]
```diff
@@ -25,7 +25,7 @@ p({
   },
   "parent":"admin"
 })
-p({"class":"foam.nanos.boot.DAONSpecMenu","id":"admin.data","label":"Data Management","handler":{"class":"foam.nanos.menu.ViewMenu","view":{"class":"foam.nanos.boot.DAOConfigSummaryView"}},"parent":"admin"})
+p({"class":"foam.nanos.boot.DAONSpecMenu","id":"admin.data","order":1,"label":"Data Management","handler":{"class":"foam.nanos.menu.ViewMenu","view":{"class":"foam.nanos.boot.DAOConfigSummaryView"}},"parent":"admin"})
```
Comment (Nauna) [inline src/menus:28]: still need to review - menu orders

File: src/foam/u2/crunch/wizardflow/ConfigureFlowAgent.js [hunk 2]
```diff
@@ -43,33 +43,22 @@ foam.CLASS({
...
-      class: 'String'
-    },
     {
       name: 'pushView',
       class: 'Function',
       expression: function () {
-        var self = this;
         return this.popupMode
-          ? function (viewSpec, onClose) {
+          ? (viewSpec, onClose) => {
```
Comment (Nauna) [inline src/foam/u2/crunch/wizardflow/ConfigureFlowAgent.js:51]: personal note : wonder why this was a function property and not a method?

File: src/foam/nanos/controller/ApplicationController.js [hunk 3]
```diff
@@ -641,7 +641,11 @@ foam.CLASS({
       }
       /** Use to load a specific menu. **/
       // Do it this way so as to not reset mementoTail if set
-      // todo ; not this will trigger another call to pushMenu ... should ensure one change by adding a check for current state and menu change....if its the same don't doo menu calls
+      // NOTE: if pushMenu is called from code the memento changes here cause memento.head !== menu
+      //       and this then changes the memento.value which then triggers ApplicationController.mementoChange
+      //       which then trigger pushMenu again.
+      //       however it only does one duplicate call and think a feedback check isn't worth it.
+      //       noneTheLess leaving topic for future consideration
```
Comment (sarthak-marwaha) [inline src/foam/nanos/controller/ApplicationController.js]: stack.push() checks for duplicate pushes already, It just requires an id property to use as a reference which I think we already add in menu.launch()

File: src/menus (hunk 1 above)
Comment (Nauna) [reply src/menus:28]: done - coincides with nanopay pr. If you have admin.data menu this makes sense to be the first landing spot.

**Pull request closed without merging**